### PR TITLE
Fix parent class of generated VSS specifications

### LINKED ISSUE #62: VSS-Processor: parentClass of generated VSS Specifications incorrect
**Is your feature request related to a problem? Please describe.**
Right now the parentClass of generated VSS Specifications always points to itself and not as it should to the class above it. 
e.g. VssHeartRate (Vehicle.Driver.HeartRate) points to "VssHeartRate.class" instead of "VssDriver.class".

**Describe the solution you'd like**
parentClass points to the real parent and not itself.
A regression test should be added, e.g. in the TestApp, where the Specifications are already available. 

**Alternate Solution**
Remove parent attribute from vss specifications

### PR DESCRIPTION
**Pull request by Chrylo**
The parent class is now correct for the property. 

- A parent inside the same file does not need an import and could easily be identified for the current class spec
- A parent in another file (Same as file name) needs an import so we iterate through past generated files to find the correct class
- A parent in another file (Sub class) needs an import so in addition to the above we add a sub import of the actual parent class.

See comments in the added search method for detailed explanations.

close #62

### CODE REVIEW

File: vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt [hunk 1]
```diff
@@ -128,11 +130,50 @@ class VssDefinitionProcessor(
...
 
                 file.writeTo(codeGenerator, false)
+                generatedFilesVssPathToClassName[vssPath.path] = className
+            }
+        }
+
+        // Uses a map of vssPaths to ClassNames which are validated if it contains a parent of the given specModel.
+        // If the actual parent is a sub class (Driver) in another class file (e.g. Vehicle) then this method returns
+        // a sub import e.g. "Vehicle.Driver". Otherwise just "Vehicle" is returned.
+        private fun createParentImport(
```
Comment (wba2hi) [inline vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt:153]: Optional:
Somehow this name does not really fit. My personal expectation is, that when something is "created" it's normally a more complex object than a String. 
Maybe something like "findParent" or similar is more appropriate.

File: vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt [hunk 2]
```diff
@@ -128,11 +130,50 @@ class VssDefinitionProcessor(
...
+            // Iterate up from the parent until the actual file name = class name was found. This indicates
+            // that the actual parent is a sub class in this file.
+            while (availableParentVssPath.contains(".")) {
+                availableParentVssPath = availableParentVssPath.substringBeforeLast(".")
+
+                parentSpecClassName = parentVssPathToClassName[availableParentVssPath]
+                if (parentSpecClassName != null) break
+            }
+
+            if (parentSpecClassName == null) return ""
```
Comment (wba2hi) [inline vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt:169]: Optional:
We have implicite knowledge, that only Vehicle should have a null parent. Should we adapt the logic, to throw an Exception or at least log an error when this occurs on any other (unexpected) VssPath?

**Review by wba2hi — CHANGES_REQUESTED** [review]
Minor findings. 

Functionality wise it looks fine! Good job.

File: vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt (hunk 2 above)
Comment (Chrylo) [reply vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt:169]: For the method itself both cases are valid. A parent was found or no parent was found (then it returns an empty string). So I don't like putting it here but I will add a log entry. Checking this case will result in an "ugly" if case. I prefer not validating this case. :)

File: vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt (hunk 1 above)
Comment (Chrylo) [reply vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/VssDefinitionProcessor.kt:153]: Renamed to "build" like in StringBuilder.

**Review by wba2hi — APPROVED** [review]
LGTM

Good work

**Pull request merged**